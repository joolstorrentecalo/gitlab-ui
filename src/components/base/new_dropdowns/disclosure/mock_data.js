export const mockItems = [
  {
    text: 'Mark as draft',
    href: 'https://gitlab.com',
    extraAttrs: {
      target: '_blank',
      rel: 'nofollow',
      'data-method': 'put',
    },
  },
  {
    text: 'Close merge request',
    action: () => {
      // eslint-disable-next-line no-console
      console.log('CLOSED');
    },
    extraAttrs: {
      class: 'gl-text-red-500!',
      rel: 'nofollow',
      'data-method': 'put',
    },
  },
  {
    text: 'Create new',
    href: 'https://gitlab.com/gitlab-org/gitlab/-/merge_requests/new',
    extraAttrs: {
      rel: 'nofollow',
      target: '_blank',
    },
  },
];

export const mockItemsCustomItem = [
  {
    text: 'Assigned to you',
    href: 'https://gitlab.com/dashboard/merge_requests?assignee_username=root',
    count: '2',
    extraAttrs: {
      target: '_blank',
      rel: 'nofollow',
    },
  },
  {
    text: 'Review requests from you',
    href: 'https://gitlab.com/dashboard/merge_requests?reviewer_username=root',
    count: 0,
    extraAttrs: {
      target: '_blank',
      rel: 'nofollow',
    },
  },
  {
    text: 'I am only visible on mobile!',
    action: () => {
      // eslint-disable-next-line no-console
      console.log('clicked!');
    },
    wrapperClass: 'gl-sm-display-none!',
  },
];

export const mockGroupsCustomItem = [
  {
    name: 'Merge requests',
    items: [
      {
        text: 'Assigned to you',
        href: 'https://gitlab.com/dashboard/merge_requests?assignee_username=root',
        count: 1,
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'Review requests from you',
        href: 'https://gitlab.com/dashboard/merge_requests?reviewer_username=root',
        count: 4,
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
    ],
  },
];

export const mockGroups = [
  {
    name: 'This project',
    items: [
      {
        text: 'New issue',
        href: 'https://gitlab.com/gitlab-org/gitlab/-/issues/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'New merge request',
        href: 'https://gitlab.com/gitlab-org/gitlab/-/merge_requests/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'New snippet',
        href: 'https://gitlab.com/gitlab-org/gitlab/-/snippets/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
    ],
  },
  {
    name: 'GitLab',
    items: [
      {
        text: 'New project',
        href: 'https://gitlab.com/projects/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'New group',
        href: 'https://gitlab.com/groups/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'New snippet',
        href: 'https://gitlab.com/snippets/new',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
    ],
  },
];

export const mockProfileGroups = [
  {
    items: [
      {
        text: 'Orange Fox',
        secondaryText: '@thefox',
        action: () => {
          window.open('https://gitlab.com/gitlab');
        },
      },
    ],
  },
  {
    items: [
      {
        text: 'Set status',
        href: 'https://gitlab.com',
        icon: 'status_success',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'Edit profile',
        href: '#',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
      {
        text: 'Preferences',
        href: '#',
        extraAttrs: {
          target: '_blank',
          rel: 'nofollow',
        },
      },
    ],
  },
  {
    items: [
      {
        text: 'Sign out',
        action: () => {
          // eslint-disable-next-line no-alert
          window.confirm('Are you sure?');
        },
      },
    ],
  },
];
