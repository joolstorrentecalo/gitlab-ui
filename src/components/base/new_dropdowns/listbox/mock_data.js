export const mockOptions = [
  {
    value: 'prod',
    text: 'Product',
  },
  {
    value: 'ppl',
    text: 'People',
  },
  {
    value: 'fin',
    text: 'Finance',
  },
  {
    value: 'leg',
    text: 'Legal',
  },
  {
    value: 'eng',
    text: 'Engineering',
  },
  {
    value: 'sales',
    text: 'Sales',
  },
  {
    value: 'marketing',
    text: 'Marketing',
  },
  {
    value: 'acc',
    text: 'Accounting',
  },
  {
    value: 'hr',
    text: 'Human Resource Management',
  },
  {
    value: 'rnd',
    text: 'Research and Development',
  },
  {
    value: 'cust',
    text: 'Customer Service',
  },
  {
    value: 'sup',
    text: 'Support',
  },
];

export const mockGroups = [
  {
    text: 'Branches',
    options: [
      { text: 'main', value: 'main' },
      { text: 'feature-123', value: 'feature-123' },
    ],
  },
  {
    text: 'Tags',
    options: [
      { text: 'v1.0', value: 'v1.0' },
      { text: 'v2.0', value: 'v2.0' },
      { text: 'v2.1', value: 'v2.1' },
    ],
  },
];
