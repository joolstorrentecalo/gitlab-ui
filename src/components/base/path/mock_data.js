export const mockPathItems = [
  {
    title: 'First',
    metric: '1d',
  },
  {
    title: 'Second',
    metric: '2d',
  },
  {
    title: 'Third',
    metric: '3d',
  },
  {
    title: 'Fourth',
    metric: '4d',
  },
  {
    title: 'Fifth',
    metric: '5d',
  },
  {
    title: 'Sixth',
    metric: '6d',
  },
  {
    title: 'Seventh',
    metric: '7d',
  },
  {
    title: 'Eighth',
    metric: '8d',
    disabled: true,
  },
  {
    title: 'Ninth',
    metric: '9d',
  },
  {
    title: 'Tenth',
    metric: '10d',
  },
];

export const mockDisclosurePathItems = [
  {
    title: 'First',
    icon: 'epic',
    href: '#',
  },
  {
    title: 'Second',
    icon: 'epic',
    href: '#',
  },
  {
    title: 'Third',
    icon: 'epic',
    href: '#',
  },
  {
    title: 'Fourth',
    icon: 'epic',
    href: '#',
  },
  {
    title: 'Fifth',
    icon: 'issues',
    href: '#',
  },
  {
    title: 'Sixth',
    icon: 'issues',
    href: '#',
  },
  {
    title: 'Seventh',
    icon: 'issues',
    href: '#',
  },
  {
    title: 'Eighth',
    icon: 'issue-type-task',
    href: '#',
    disabled: true,
  },
  {
    title: 'Ninth',
    icon: 'issue-type-task',
    href: '#',
  },
  {
    title: 'Tenth',
    icon: 'issue-type-task',
    href: '#',
  },
];
