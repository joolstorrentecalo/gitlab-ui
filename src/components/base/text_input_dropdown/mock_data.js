export const GROUPED_ITEMS = [
  {
    group: 'Issues',
    items: [
      { text: 'Lorem ipsum', iid: 1, type: 'issue' },
      { text: 'Dolar sit', iid: 2, type: 'issue' },
      { text: 'Amit', iid: 3, type: 'issue' },
      { text: 'Consectetur adipiscing', iid: 4, type: 'issue' },
      { text: 'Elit sed', iid: 5, type: 'issue' },
    ],
  },
  {
    group: 'Merge Requests',
    items: [
      { text: 'Do eiusmod', iid: 1, type: 'mr' },
      { text: 'Tempor incididunt', iid: 2, type: 'mr' },
      { text: 'Ut labore', iid: 3, type: 'mr' },
      { text: 'Et dolore', iid: 4, type: 'mr' },
      { text: 'Magna aliqua', iid: 5, type: 'mr' },
    ],
  },
  {
    group: 'Projects',
    items: [
      { text: 'Ut enim', iid: 1, type: 'project' },
      { text: 'Ad minim', iid: 2, type: 'project' },
      { text: 'Veniam quis', iid: 3, type: 'project' },
      { text: 'Nostrud exercitation', iid: 4, type: 'project' },
      { text: 'Ullamco laboris', iid: 5, type: 'project' },
    ],
  },
];

export const filterGroups = (text) => {
  if (!text) {
    return GROUPED_ITEMS;
  }

  return (
    GROUPED_ITEMS
      // Filter all items of groups
      .map(({ items, ...rest }) => ({
        items: items.filter((x) => x.text.toLowerCase().includes(text.toLowerCase())),
        ...rest,
      }))
      // Filter out groups with no matching items
      .filter((x) => x.items.length)
  );
};
