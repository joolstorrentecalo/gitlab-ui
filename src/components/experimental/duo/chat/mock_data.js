import { DOCUMENTATION_SOURCE_TYPES, MESSAGE_MODEL_ROLES } from './constants';

const MOCK_SOURCES = [
  {
    title: 'GitLab Handbook',
    source_type: DOCUMENTATION_SOURCE_TYPES.HANDBOOK.value,
    source_url: '/handbook/',
  },
  {
    stage: 'Mock Stage',
    group: 'Mock Group',
    source_type: DOCUMENTATION_SOURCE_TYPES.DOC.value,
    source_url: '/company/team/',
  },
  {
    date: '2023-04-21',
    author: 'Test User',
    source_type: DOCUMENTATION_SOURCE_TYPES.BLOG.value,
    source_url: '/blog/',
  },
];

export const MOCK_RESPONSE_MESSAGE = {
  id: '123',
  content: '_Duo Chat message_ comming from AI',
  contentHtml: '<p><em>Duo Chat message</em> comming from AI</p>',
  role: MESSAGE_MODEL_ROLES.assistant,
  extras: {
    sources: MOCK_SOURCES,
  },
  requestId: '987',
  errors: [],
  timestamp: '2021-04-21T12:00:00.000Z',
};
