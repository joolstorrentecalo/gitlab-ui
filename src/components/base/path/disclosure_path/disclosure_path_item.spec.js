import { shallowMount } from '@vue/test-utils';
import { mockDisclosurePathItems } from '../mock_data';
import GlDisclosurePathItem from './disclosure_path_item.vue';

describe('DisclosurePathItem', () => {
  let wrapper;

  const createComponent = (props = {}, options = {}) => {
    return shallowMount(GlDisclosurePathItem, {
      propsData: {
        item: mockDisclosurePathItems[0],
        ...props,
      },
      ...options,
    });
  };

  beforeEach(() => {
    wrapper = createComponent();
  });

  describe('renders the item', () => {
    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the inline icon', () => {
      const icon = wrapper.find('[data-testid="gl-disclosure-path-item-icon"]');

      expect(icon.exists()).toBe(true);
      expect(icon.props('name')).toBe(mockDisclosurePathItems[0].icon);
    });
  });

  describe('item slot', () => {
    beforeEach(() => {
      wrapper = createComponent(null, {
        scopedSlots: {
          default: `
            <div
              data-testid="path-item-slot-content">
              {{ props.pathItem.title }}
            </div>
          `,
        },
      });
    });

    it('contains all elements passed into the additional slot', () => {
      const pathItem = wrapper.find('[data-testid="path-item-slot-content"]');

      expect(pathItem.text()).toBe(mockDisclosurePathItems[0].title);
    });
  });
});
