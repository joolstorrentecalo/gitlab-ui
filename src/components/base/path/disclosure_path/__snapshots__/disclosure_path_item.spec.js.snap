// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisclosurePathItem renders the item matches the snapshot 1`] = `
<li
  class="gl-path-disclosure-list-item"
>
  <button
    class="gl-disclosure-path-button"
  >
    <gl-icon-stub
      class="gl-disclosure-path-item-icon gl-mx-2 gl-text-gray-600"
      data-testid="gl-disclosure-path-item-icon"
      name="epic"
      size="16"
    />
    <span
      class="gl-disclosure-path-button-content"
    >
      First
    </span>
  </button>
</li>
`;
